anvil/tests/it/main.rs

```diff
@@ -20,16 +17,19 @@ mod txpool;
 pub mod utils;
 mod wsapi;
 
-// keeps track of ports that can be used
-pub static NEXT_PORT: AtomicU16 = AtomicU16::new(8546);
+/// Binds to port 0, asking for a random TCP port.
+///
+/// Returns None if the OS did not give us a port.
+fn ask_free_tcp_port() -> Option<u16> {
+    // Binding to port 0 asks the OS to assign us a random port.
```

Comment:
Just a note here, this is the case on all operating systems. On Windows it is documented by Microsoft in the docs for `bind`, on Linux it's undocumented but the behavior has been there forever so I think they just consider it to be a trivial detail

Reply:
yeh, I need to rewrite some things first to get the address back when we bind the anvil server